best-practices/sdk-principles/kotlin.md

```diff
@@ -0,0 +1,73 @@
+# Client software (SDK) principles and best practice for Kotlin (and any other JVM language)
+
+Background and product stratagy context can be found in [SDK principles for JS](js.md)
+
+Kotlin is the primary development language for any JVM-based SDKs (Java, Scala, Groovy, Clojure, etc). Our main focus is to provide an idiomatic Kotlin API for our customers. However, upon request, we will also offer a Java package that does not use coroutines or extension functions, making it easier to consume from Java (similar to [apollo java support](https://github.com/apollographql/apollo-kotlin-java-support)).
+
+## Core SDK
+
+The Core SDK serves is a Pub/Sub channel SDK without additional plugins like VcDiff, LiveObjects, or Push Notifications.
+
+Our current pub/sub channel SDK should be split into several modules:
```

Comment:
This seems more like a vision/goal instead of what we currently have?

Reply:
Yep, added current state, rename section to the "future improvements"